### PR TITLE
Hamza

### PR DESCRIPTION
**Pull request by hamzashariq7**
Added devise and migrations/models.

### CODE REVIEW

**Comment by hamzashariq7** [comment]
cycle 0.

File: Gemfile [hunk 1]
```diff
@@ -20,6 +20,8 @@ gem 'jbuilder', '~> 2.0'
 # bundle exec rake doc:rails generates the API under doc/api.
 gem 'sdoc', '~> 0.4.0', group: :doc
 
+gem 'turbolinks'
```
Comment (zohaibbutt13) [inline Gemfile]: remove gem turbolinks

File: app/models/company.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Company < ActiveRecord::Base
+	belongs_to :subscription_package
```
Comment (zohaibbutt13) [inline app/models/company.rb]: indentation

File: app/models/company.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+	has_one :company_setting
+	has_many :groups
+	has_many :users
```
Comment (zohaibbutt13) [inline app/models/company.rb]: what about surveys, questions, responses and other tables?

File: app/models/group.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+class Group < ActiveRecord::Base
+	belongs_to :company
+	has_many :group_members
```
Comment (zohaibbutt13) [inline app/models/group.rb]: what is group_member? another module? why need this?

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<p>Welcome <%= @email %>!</p>
```
Comment (zohaibbutt13) [inline app/views/devise/mailer/confirmation_instructions.html.erb]: use full name instead of email

File: app/models/group.rb (hunk 4 above)
Comment (hamzashariq7) [reply app/models/group.rb]: Needed to store which member is in which group inside database. contains group_id, user_id pairs.

File: app/models/company.rb (hunk 3 above)
Comment (hamzashariq7) [reply app/models/company.rb]: Surveys belong to users (who created them). Questions belong to surveys. Responses belong to questions.

File: Gemfile (hunk 1 above)
Comment (hamzashariq7) [reply Gemfile]: ok

File: app/views/devise/passwords/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<h2>Forgot your password?</h2>
+
+<%= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :post }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :email %><br />
```
Comment (zohaibbutt13) [inline app/views/devise/passwords/new.html.erb]: use for attribute

File: app/views/devise/registrations/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,29 @@
+<h2>Sign up</h2>
```
Comment (zohaibbutt13) [inline app/views/devise/registrations/new.html.erb]: can we use single partial for new and edit actions ?

File: config/database.yml [hunk 8]
```diff
@@ -1,13 +1,31 @@
-development:
```
Comment (zohaibbutt13) [inline config/database.yml]: discuss with me

File: db/migrate/20190716113519_devise_create_users.rb [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
+
+      t.timestamps null: false
+    end
+
+    add_index :users, :email,                unique: true
+    add_index :users, :reset_password_token, unique: true
+    # add_index :users, :confirmation_token,   unique: true
```
Comment (zohaibbutt13) [inline db/migrate/20190716113519_devise_create_users.rb]: why comment this ?

File: db/migrate/20190716125356_add_fields_to_users.rb [hunk 10]
```diff
@@ -0,0 +1,8 @@
+class AddFieldsToUsers < ActiveRecord::Migration
+  def change
+    add_reference :users, :company, index: true
+    add_column :users, :f_name, :string
```
Comment (zohaibbutt13) [inline db/migrate/20190716125356_add_fields_to_users.rb]: user proper name for fields f_name -> first_name

File: db/migrate/20190716113519_devise_create_users.rb (hunk 9 above)
Comment (hamzashariq7) [reply db/migrate/20190716113519_devise_create_users.rb]: Fixed.

File: db/migrate/20190716125356_add_fields_to_users.rb (hunk 10 above)
Comment (hamzashariq7) [reply db/migrate/20190716125356_add_fields_to_users.rb]: Fixed.

File: app/models/group.rb [hunk 11]
```diff
@@ -0,0 +1,7 @@
+class Group < ActiveRecord::Base
+  belongs_to :company
+  has_many :group_members
+
+  validates :name, presence: true, uniqueness: true, length: { maximum: 20 }
```
Comment (zohaibbutt13) [inline app/models/group.rb]: change length to 50

**Pull request merged**